nixos/modules/services/networking/aria2.nix

```diff
@@ -11,86 +9,122 @@ let
+        one of many WebUIs (http://localhost:6800/ by default).
+
+        Targets are downloaded to ${downloadDir} by default and are
+        accessible to users in the "aria2" group.
+      ''
+      );
+      openPorts = lib.mkOption {
+        type = lib.types.bool;
         default = false;
         description = lib.mdDoc ''
           Open listen and RPC ports found in listenPortRange and rpcListenPort
           options in the firewall.
         '';
       };
-      downloadDir = mkOption {
-        type = types.path;
+      downloadDir = lib.mkOption {
+        type = lib.types.path;
         default = downloadDir;
         description = lib.mdDoc ''
           Directory to store downloaded files.
         '';
       };
-      listenPortRange = mkOption {
-        type = types.listOf types.attrs;
-        default = [ { from = 6881; to = 6999; } ];
+      listenPortRange = lib.mkOption {
+        type = lib.types.listOf lib.types.attrs;
+        default = [{ from = 6881; to = 6999; }];
         description = lib.mdDoc ''
           Set UDP listening port range used by DHT(IPv4, IPv6) and UDP tracker.
         '';
       };
-      rpcListenPort = mkOption {
-        type = types.int;
+      rpcListenPort = lib.mkOption {
+        type = lib.types.int;
         default = 6800;
         description = lib.mdDoc "Specify a port number for JSON-RPC/XML-RPC server to listen to. Possible Values: 1024-65535";
       };
-      rpcSecretFile = mkOption {
-        type = types.path;
+      rpcSecretFile = lib.mkOption {
+        type = lib.types.path;
         example = "/run/secrets/aria2-rpc-token.txt";
         description = lib.mdDoc ''
           A file containing the RPC secret authorization token.
           Read https://aria2.github.io/manual/en/html/aria2c.html#rpc-auth to know how this option value is used.
         '';
       };
-      extraArguments = mkOption {
-        type = types.separatedString " ";
-        example = "--rpc-listen-all --remote-time=true";
-        default = "";
+      settings = lib.mkOption {
         description = lib.mdDoc ''
-          Additional arguments to be passed to Aria2.
+          Generates the `aria2.conf` file. Refer to [the documentation][0] for
+          all possible settings. The following settings are implicitly set by
+          this module: `dir`, `listen-port`, `rpc-listen-port`, `enable-rpc`,
+          `save-session` and `conf-path`
+
+          [0]: https://aria2.github.io/manual/en/html/aria2c.html#synopsis
         '';
+        default = { };
+        type = lib.types.submodule {
+          freeformType = with lib.types; attrsOf (oneOf [ singleLineStr int float bool ]);
+        };
       };
     };
   };
 
-  config = mkIf cfg.enable {
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.settings ? dir);
+        message = "Use `services.aria2.downloadDir` to set the download directory.";
+      }
```

Comment:
Can we `mkRenamedOptionModule` to remove `downloadDir` and use `settings.dir` instead?